### PR TITLE
Clarify that label filter expressions are not supported in promtail.

### LINKED ISSUE #5209: Label selectors in LogQL work in loki, but not in promtail match
**Describe the bug**
I am trying to put a match selector in promtail to select by a range of http status codes. I am using this line here:

    - match: 
      selector: '{status=~".+",level !~ ".+"} | status>=100 status<=399'

This query standalone works fine in grafana/loki UI, but when using this as a selector in promtail config, I get this error:
`invalid match stage config: invalid selector syntax for match stage: parse error at line 1, col 28: syntax error: unexpected IDENTIFIER, expecting != or !~ or |~ or |="`

Was also documented on grafana forums by someone else: [https://community.grafana.com/t/promtail-2-3-0-label-filtering-no-longer-works-with-match-stage/52435](https://community.grafana.com/t/promtail-2-3-0-label-filtering-no-longer-works-with-match-stage/52435)

**To Reproduce**
1. Started Promtail daemonset - image grafana/loki:2.4.2
2. Ensure a match selector includes a label filter
3. Daemonset fails

**Expected behavior**
Queries that work in logQL in Grafana UI should work as logQL selectors, or if their are differences in operation they should be documented.

**Environment:**
 - Infrastructure: k3s, kubernetes daemonset
 - Deployment tool: yaml

### PR DESCRIPTION
**Pull request by jeschkies**
**What this PR does / why we need it**:
Some users of Promtail seem to be confused that certain types of filter expressions are not supported in Promtail's `match` stage.

This change clarifies what kind of filter expressions are supported.

**Which issue(s) this PR fixes**:
Fixes #5209

**Special notes for your reviewer**:

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: docs/sources/clients/promtail/stages/match.md [hunk 1]
```diff
@@ -5,14 +5,18 @@ description: match stage
 # match
 
 The match stage is a filtering stage that conditionally applies a set of stages
-or drop entries when a log entry matches a configurable [LogQL]({{< relref "../../../query" >}})
-stream selector and filter expressions.
+or drop entries when a log entry matches a configurable LogQL
+[stream selector]({{< relref "../../../query/log_queries/#log-stream-selector" >}}) and
```
Comment (JStickler) [inline docs/sources/clients/promtail/stages/match.md:9]:
```suggestion
[stream selector]({{< relref "../../../query/log_queries#log-stream-selector" >}}) and
```

File: docs/sources/clients/promtail/stages/match.md [hunk 2]
```diff
@@ -5,14 +5,18 @@ description: match stage
...
+[filter expressions]({{< relref "../../../query/log_queries/#line-filter-expression" >}}).
```
Comment (JStickler) [inline docs/sources/clients/promtail/stages/match.md:10]:
```suggestion
[filter expressions]({{< relref "../../../query/log_queries#line-filter-expression" >}}).
```

**Review by JStickler — COMMENTED** [review]
[docs team]

File: docs/sources/clients/promtail/stages/match.md [hunk 3]
```diff
@@ -5,14 +5,18 @@ description: match stage
...
 
 The match stage is a filtering stage that conditionally applies a set of stages
-or drop entries when a log entry matches a configurable [LogQL]({{< relref "../../../query" >}})
-stream selector and filter expressions.
+or drop entries when a log entry matches a configurable LogQL
+[stream selector]({{< relref "../../../query/log_queries#log-stream-selector" >}}) and
+[filter expressions]({{< relref "../../../query/log_queries#line-filter-expression" >}}).
+
+Please note that the filters do not include label filter expressions such as `|
+label == "foobar"`.
```
Comment (JStickler) [inline docs/sources/clients/promtail/stages/match.md:13]:
```suggestion
```suggestion
{{% admonition type="note" %}}
The filters do not include label filter expressions such as `| label == "foobar"`.
{{% /admonition %}}



```

**Review by JStickler — APPROVED** [review]
[docs team]  LGTM

File: docs/sources/clients/promtail/stages/match.md [hunk 4]
```diff
@@ -5,14 +5,20 @@ description: match stage
 # match
 
 The match stage is a filtering stage that conditionally applies a set of stages
-or drop entries when a log entry matches a configurable [LogQL]({{< relref "../../../query" >}})
-stream selector and filter expressions.
+or drop entries when a log entry matches a configurable LogQL
+[stream selector]({{< relref "../../../query/log_queries#log-stream-selector" >}}) and
+[filter expressions]({{< relref "../../../query/log_queries#line-filter-expression" >}}).
+
+```suggestion
```
Comment (chaudum) [inline docs/sources/clients/promtail/stages/match.md:12]: This line looks like an unwanted artifact. Is it?

**Review by chaudum — CHANGES_REQUESTED** [review]

File: docs/sources/clients/promtail/stages/match.md (hunk 4 above)
Comment (jeschkies) [reply docs/sources/clients/promtail/stages/match.md]: Thanks for catching. GitHub must have applied a suggestion wrong.

**Pull request merged**